src/data.py

```diff
@@ -8,258 +8,372 @@
 from utils import split_chunks, RangeDict
 from collections import defaultdict
 
-from transformers import AutoTokenizer, DataCollatorForTokenClassification
+from transformers import DataCollatorForTokenClassification
 
-ENTITIES = ['SYMPTOM', 'PROCEDURE', 'DISEASE', 'PROTEIN', 'CHEMICAL']
+from itertools import groupby
+from operator import itemgetter
 
-def load_train_val( tokenizer,
-                    context_size=64,
-                    test_split_percentage=0.15,
-                    train_transformations=None,
-                    train_augmentations=None,
-                    entity= ['SYMPTOM', 'PROCEDURE', 'DISEASE', 'PROTEIN', 'CHEMICAL'],
-                    test_transformations=None
-                    ):
+# import logging
 
-    global ENTITIES
-    ENTITIES = entity
-    docs = defaultdict(list)
-    
-    train = '../dataset/merged_data_subtask1_train.tsv'
 
-    df = pd.read_csv(train, sep="\t")
-    df = df[df.label.isin(entity)]
+# logger = logging.getLogger()
+# if logger.hasHandlers():
+#     logger.handlers.clear()
+# logger.setLevel(logging.DEBUG)
+# stream_handler = logging.StreamHandler()
+# formatter = logging.Formatter('%(asctime)s - %(levelname)s - %(message)s')
+# stream_handler.setFormatter(formatter)
+# logger.addHandler(stream_handler)
 
-    for _, row in df.iterrows():
-        docs[row["filename"]].append({k:row[k] for k in ["ann_id", "label", "start_span", "end_span", "text"]})
-    
+
+class Corpus:
+    #  A class that represent a corpus. If you want a specific dataset, you need to implement a function that returns a corpus
+    def __init__(self, data: list ):
```

Comment:
Explain the format of the data argument should be in the readme, for me it seems the main entry point to add other datasets. If your data fits the that format it will always work right?

Reply:
line 36: #Data needs to be formated as follows {"doc_id":id, "text": doucment_text, "annotations":[{'label': LABEL, 'start_span': num, 'end_span': num}, ... ]  }   
I agree should be in Readme.  But yes if the data is in the same format all other funcitons work.